duckduckgo/Android
# Experiment with enabling prompted Fireproofing by default

Task/Issue URL: https://app.asana.com/0/488551667048375/1199920127722390/f
Tech Design URL: https://app.asana.com/0/1142021229838617/1199920405912714/f
CC: 

**Description**:
We want to run an experiment to measure the effect of enabling Fireproof login detection by default.

**Control group**: no changes.

**Experimental variant**: we've introduced the following product changes:
- Fireproof login detection enabled by default
- Allow users to disable login detection if they dismiss twice in a row the Fireproof login prompt dialog
- Introduce pixels as defined in https://app.asana.com/0/0/1199699295733011/f

**Steps to test this PR**:

Control group:
1. Hardcode variant "zq" 
1. Fresh intall
2. Go to settings and ensure login detection is **disabled**
3. Enable login detection
4. Perform login in **two** different websites, and dismiss the Fireproof login dialog in all of them
5. Ensure we never asked you to disable login detection

Experiment Test 1: when Login dialog dismissed twice in a row, ask user to disable login detection
1. Hardcode variant "zw" 
1. Fresh intall
2. Go to settings and ensure login detection is **enabled**
3. Go back to the browser
4. Perform login in **two** different websites, and dismiss the Fireproof login dialog in all of them
5. Ensure dialog asking to disable login detection shown
6. Press disable
7. Go to fireproof settings and ensure login detection is disabled

Experiment Test 2: we ask to disable login detection only once
1. Hardcode variant "zw" 
1. Fresh intall
4. Perform login in **two** different websites, and dismiss the Fireproof login dialog in all of them
5. Ensure dialog asking to disable login detection shown
6. Press "Cancel"
7. Clear all data (this is optional, clear data so we can login on sites again)
4. Perform login in **two** different websites, and dismiss the Fireproof login dialog in all of them
5. Ensure we don't show again the dialog to disable login detection

Experiment Test 3: if user enables login detection, never ask them to disable login detection
1. Hardcode variant "zw" 
1. Fresh intall
2. Go to Settings
3. Ensure login detection is enabled
4. Disable login detection
5. Enable it again
4. Perform login in **two** different websites, and dismiss the Fireproof login dialog in all of them
5. Ensure we don't show again the dialog to disable login detection

Experiment Test 4: Validate pixels (user did not try the fire button)
1. Hardcode variant "zw" 
1. Fresh intall
4. Perform login on a website
5. Ensure Fireproof login dialog shown and pixel `m_fw_ld_s` with `fe=false`
6. Dismiss dialog and pixel `m_fw_l_d` with `fe=false`
7. Do login on a second site and dismiss Fireproof login dialog
5. Ensure dialog asking to disable login detection shown and pixel `m_fw_dd_s` with `fe=false`
5. Dismiss dialog and pixel `m_fw_dd_c` with `fe=false`
6. Do login on a third site and when asked to fireproof website click "Fireproof"
7.  Dialog dismissed and pixel `m_fw_l_a` with `fe=false`sent

Experiment Test 5: Validate pixels (user already tried the fire button)
1. Hardcode variant "zw" 
1. Fresh intall
2. Clear all data
4. Perform login on a website
5. Ensure Fireproof login dialog shown and pixel `m_fw_ld_s` with `fe=true`
6. Dismiss dialog and pixel `m_fw_l_d` with `fe=true`
7. Do login on a second site and dismiss Fireproof login dialog
5. Ensure dialog asking to disable login detection shown and pixel `m_fw_dd_s` with `fe=true`
5. Dismiss dialog and pixel `m_fw_dd_c` with `fe=true`
6. Do login on another site and when asked to fireproof website click "Fireproof"
7.  Dialog dismissed and pixel `m_fw_l_a` with `fe=true` sent


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
